### PR TITLE
improved interpolation for FL33

### PR DESCRIPTION
**Pull request by Thomas-Ulrich**

### CODE REVIEW

**Comment by fabian-kutschera** [comment]
Hi Thomas,
methods `linear `and `nearest ` result in the correct slip value in the end.
`slinear`, `cubic `and `quintic ` seem not defined yet and do not work.

Additional comments:

- You might want to consider to move `--spatial_zoom ` from the optional to the required arguments
- The output of -h does not give any information which interpolation methods would be available. Perhaps it would be useful to print the choices as well.

> python generate_FL33_input_files.py -h
> usage: generate_FL33_input_files.py [-h] [--interpolation_method interpolation_method]
>                                     --spatial_zoom spatial_zoom [--write_paraview]
>                                     [--generate_ts_yaml proj] [--PSRthreshold PSRthreshold]
>                                     filename
> 
> generate yaml and netcdf input to be used with friction law 33/34 based on a (hereupsampled)
> kinematic model in the standard rupture format srf file.
> 
> positional arguments:
>   filename              filename of the srf file
> 
> optional arguments:
>   -h, --help            show this help message and exit
>   --interpolation_method interpolation_method
>                         interpolation method
>   --spatial_zoom spatial_zoom
>                         level of spatial upsampling
>   --write_paraview      write also netcdf readable by paraview
>   --generate_ts_yaml proj
>                         generate fault yaml file and write fault geometry (ts file). proj:
>                         proj4 string describing the projection.
>   --PSRthreshold PSRthreshold
>                         peak slip rate threshold (0-1) to determine onset time and duration of
>                         STF
> 
>

**Comment by Thomas-Ulrich** [comment]
You need to update scipy.
pip3 install scipy --upgrade

Le 19 novembre 2022 12:17:54 GMT+01:00, Fabian Kutschera ***@***.***> a écrit :

**Comment by fabian-kutschera** [comment]
Now I can confirm that all of the methods work and have valid results with only minor variations (+- 0.02) in terms of absolute slip among the different choices.

File: preprocessing/science/kinematic_models/FaultPlane.py [hunk 1]
```diff
@@ -537,7 +533,81 @@ def compute_corrected_slip_for_differing_area(self, proj):
...
         return slip1
 
-    def generate_netcdf_fl33(self, prefix, spatial_order, spatial_zoom, proj, write_paraview):
+    def compute_1d_dimension_arrays(self, spatial_zoom):
+        self.spatial_zoom = spatial_zoom
+        # Compute dimension arrays
+        km2m = 1e3
+        coords = np.array([self.x, self.y, -km2m * self.depth])
+        ny, nx = coords.shape[1:3]
+        center_raw = coords[:, (ny - 1) // 2, :] - coords[:, (ny - 1) // 2 - 1, :]
```
Comment (sebwolf-de) [inline preprocessing/science/kinematic_models/FaultPlane.py:542]: Probably, you mean row instead of raw?

Comment (Thomas-Ulrich) [reply preprocessing/science/kinematic_models/FaultPlane.py]: yes

**Review by davschneller — APPROVED** [review]

**Pull request merged**